File: apps/api/schemas/meeting.py
from typing import Literal

from pydantic import BaseModel, ConfigDict


class Participant(BaseModel):
    model_config = ConfigDict(extra="forbid")

    name: str
    role: str | None
    evidence: str
    certainty: Literal["explicit", "inferred"]


class Topic(BaseModel):
    model_config = ConfigDict(extra="forbid")

    title: str
    summary: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class DiscussionPoint(BaseModel):
    model_config = ConfigDict(extra="forbid")

    topic: str
    summary: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class Decision(BaseModel):
    model_config = ConfigDict(extra="forbid")

    description: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class Action(BaseModel):
    model_config = ConfigDict(extra="forbid")

    description: str
    assignee: str | None
    deadline: str | None
    evidence: str
    certainty: Literal["explicit", "inferred"]


class Problem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    description: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class OpenQuestion(BaseModel):
    model_config = ConfigDict(extra="forbid")

    question: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class ImportantInformation(BaseModel):
    model_config = ConfigDict(extra="forbid")

    content: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class TimelineEvent(BaseModel):
    model_config = ConfigDict(extra="forbid")

    date_or_reference: str
    event: str
    evidence: str
    certainty: Literal["explicit", "inferred"]


class MeetingAnalysis(BaseModel):
    model_config = ConfigDict(extra="forbid")

    language: str
    summary: str
    topics: list[Topic]
    participants: list[Participant]
    discussion: list[DiscussionPoint]
    decisions: list[Decision]
    actions: list[Action]
    problems: list[Problem]
    open_questions: list[OpenQuestion]
    important_information: list[ImportantInformation]
    timeline: list[TimelineEvent]

File: apps/web/types/meeting.ts
export type Participant = {
    name: string
    role: string | null
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type Topic = {
    title: string
    summary: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type DiscussionPoint = {
    topic: string
    summary: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type Decision = {
    description: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type Action = {
    description: string
    assignee: string | null
    deadline: string | null
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type Problem = {
    description: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type OpenQuestion = {
    question: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type ImportantInformation = {
    content: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type TimelineEvent = {
    date_or_reference: string
    event: string
    evidence: string
    certainty: "explicit" | "inferred"
  }
  
  export type MeetingAnalysis = {
    language: string
    summary: string
    topics: Topic[]
    participants: Participant[]
    discussion: DiscussionPoint[]
    decisions: Decision[]
    actions: Action[]
    problems: Problem[]
    open_questions: OpenQuestion[]
    important_information: ImportantInformation[]
    timeline: TimelineEvent[]
  }
  
  export type PreparedAction = {
    description: string
    assignee: string | null
    deadline: string | null
    status: "ready" | "needs_information"
  }
  
  export type UploadMeetingResponse = {
    filename: string
    transcript: string
    analysis: MeetingAnalysis
    actions: PreparedAction[]
  }